rfc-086-draft-stack-rummager.md

```diff
@@ -0,0 +1,37 @@
+# Draft Stack Rummager
+
+## Summary
+
+RFC 86 proposes setting up a instance of [Rummager][rummager] for the
+[draft stack][draft-stack-docs].
+
+## Problem
+
+The [draft stack (content preview)][draft-stack-docs] part of GOV.UK
+relies in places on the [rummager service (Search API)][rummager].
+
+There is only one instance of this service, which is primarily used
+for the live stack, but is also currently used from some services on
+the draft stack (see [govuk-puppet][puppet-frontend-rummager-config]).
+
+Rummager may return some draft content, depending on the internal
+index which is used. However, the amount of draft content available
+through rummager is set to decrease, as more types of content are
+migrated to the `govuk` index (see [ADR 4][rummager-adr-4]).
+
+Missing draft content makes the draft stack less useful. Previewing
+pages that use rummager for some of the data won't give an accurate
+representation of what the page would look like if the relevant
+content is published.
+
+## Proposal
+
+The proposal made here to improve the user experience on the draft
+stack, is to setup a draft instance of rummager, populated with the
+appropriate content via the Publishing API, and then switch to using
+this from draft services.
+
```

Comment:
Might be worth adding a section why this will not be implemented anytime soon:

- This depends on all content being indexed by rummager via the message queue
- Functionality needs to be added to the publishing-api to broadcast changes in drafts, including a hard think about access-limiting (we don't want sensitive documents sent to search)

Reply:
I was unsure whether it would be good to put some discussion in the RFC, or keep the discussion separate. I'm happy to put something in if you think it would help through?

Reply:
I think it's good to give some context around it. Maybe rephrase it as "implementation notes" or something like that?

Reply:
It's also worth noting that *publishing apps* have never sent to draft content to production rummager, and this has always been an unintentional thing. Rummager isn't able to distinguish between draft/live. The new indexing method is the same as the old in that respect.

Draft taxons (or links to them) are only returned by rummager now because of a stopgap measure we put in last year to keep links in sync with the publishing api, before we started working on indexing everything from the message queue. 

This is because it's using the HTTP API, which does return links to draft content, but  publishing api normally excludes these when pushing stuff to the content store/message queue.